hydroshare/hydroshare
# 4535 Remove netcdf resource type

**NOTE: This PR replaces closed PR #4528** 
 
**NOTE**: This PR has cleanup code only for NetCDF resource type. There are 10 deprecated resource types that need to deleted.  There will be separate PR for each of the deprecated resource types. This PR has been created to get early feedback as similar code cleanup strategy will be applied to other deprecated resource types. The goal is to completely remove code for deprecated resource types. As part of this PR, we won't be able to completely remove all code for a deprecated resource type as we need the resource type migrations to cleanup the database tables associated with the deprecated resource type. After the code for this PR is deployed and necessary migrations are applied, we will have another PR to completely remove the remaining code for the deprecated resource types.

### Pull Request Checklist: 















### Positive Test Case
1. Follow the following steps to locally test this:
2. Checkout develop branch and run hydroshare.
3. Create a composite resource. Create a netcdf aggregation in that resource. Verify that the aggregation has metadata. You can  add additional metadata like keywords and extended metadata (key/value).
4. Checkout the branch '4535-remove-netcdf-resource-type'. Run `./hsctl managepy migrate`.
5. Verify that you can see netcdf aggregation and its metadata. Verify that you can edit metadata for the aggregation. 
6. Verify you can create new netcdf aggregation.
